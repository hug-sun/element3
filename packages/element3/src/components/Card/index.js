import ElCard from './src/Card.vue'

/* istanbul ignore next */
ElCard.install = function (app) {
  app.component(ElCard.name, ElCard)
}

export { ElCard }
