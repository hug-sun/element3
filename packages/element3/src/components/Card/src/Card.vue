<template>
  <div class="el-card" :class="`is-${shadow}-shadow`">
    <div class="el-card__header" v-if="$slots.header || header">
      <slot name="header">{{ header }}</slot>
    </div>
    <div class="el-card__body" :style="bodyStyle">
      <slot />
    </div>
  </div>
</template>

<script lang="ts">
import { props } from './props'
import { defineComponent } from 'vue'
export default defineComponent({
  name: 'ElCard',
  props
})
</script>
