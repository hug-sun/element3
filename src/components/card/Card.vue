<script setup lang="ts">
import { withDefaults } from 'vue'
import type { StyleValue } from 'vue'
type CardType = 'always' | 'hover' | 'never'

interface CardProps {
  header?: String
  bodyStyle?: StyleValue
  shadow?: CardType
}

withDefaults(defineProps<CardProps>(), {
  shadow: 'always',
})
</script>

<template>
  <div
    class="el-card"
    :class="`is-${shadow}-shadow`"
  >
    <div v-if="$slots.header || header" class="el-card__header">
      <slot name="header">
        {{ header }}
      </slot>
    </div>
    <div class="el-card__body" :style="bodyStyle">
      <slot />
    </div>
  </div>
</template>

<style lang="scss">
@import '../../theme/src/card.scss';
</style>
