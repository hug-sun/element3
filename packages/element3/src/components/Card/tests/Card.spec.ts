import Card from '../src/Card.vue'
import { mount } from '@vue/test-utils'
import {
  expectHaveAttribute,
  expectHaveClass,
  expectHaveTextContent
} from '../../../../tests/helper'

describe('Card.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(Card)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should show content', () => {
    const content = 'body'

    const wrapper = mount(Card, {
      slots: {
        default: content
      }
    })

    expectHaveTextContent(wrapper, content)
  })

  it('shadow', async () => {
    const wrapper = mount(Card)

    expectHaveClass(wrapper, 'is-always-shadow')

    await wrapper.setProps({ shadow: 'hover' })
    expectHaveClass(wrapper, 'is-hover-shadow')

    await wrapper.setProps({ shadow: 'never' })
    expectHaveClass(wrapper, 'is-never-shadow')

    await wrapper.setProps({ shadow: 'always' })
    expectHaveClass(wrapper, 'is-always-shadow')
  })

  it('header props', async () => {
    const content = 'header'
    const wrapper = mount(Card)

    await wrapper.setProps({ header: content })
    expectHaveTextContent(wrapper.find('.el-card__header'), content)
  })

  it('header slot', () => {
    const content = 'header'

    const wrapper = mount(Card, {
      slots: {
        header: content
      }
    })

    expectHaveTextContent(wrapper.find('.el-card__header'), content)
  })

  it('body-style', () => {
    const wrapper = mount(Card, {
      props: {
        bodyStyle: {
          padding: '30px'
        }
      }
    })

    expectHaveAttribute(
      wrapper.find('.el-card__body'),
      'style',
      'padding: 30px;'
    )
  })
})
