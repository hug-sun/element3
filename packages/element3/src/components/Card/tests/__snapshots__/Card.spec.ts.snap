// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue snapshot 1`] = `
<div
  class="el-card is-always-shadow"
>
  <!--v-if-->
  <div
    class="el-card__body"
    style="padding: 20px;"
  >
    
    
  </div>
</div>
`;
